### DISCUSSION TITLE
`PytestUnknownMarkWarning` when used with `__getattr__` and `pytest.mark`

### DISCUSSION THREAD

**Discussion by sylvorg** [Textualize/rich#3362] [Q&A] [answered]
Hello!

As in the (poorly worded) title; when I attempt to use `__getattr__` and `pytest.mark`, `rich.pretty` triggers a `PytestUnknownMarkWarning`:

```python
tests/test_bundle.py: 20 warnings
  ../rich/pretty.py:631: PytestUnknownMarkWarning: Unknown pytest.mark.awehoi234_wdfjwljet234_234wdfoijsdfmmnxpi492 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    fake_attributes = hasattr(
```

Problem is, since there's no traceback, I have no idea how to even _begin_ to debug this, as I disabled every function where `__getattr__` accesses `pytest.mark`. If anyone else has hit this snag before, I would greatly appreciate any advice!

Thank you kindly for the help!

**Comment by willmcgugan** [comment, marked as answer]
That long string of gibberish is a part of a [check](https://github.com/Textualize/rich/blob/master/rich/pretty.py#L635) to see if an object is customizing `__getattr__`, which tends to break any kind of inspection.

I can't say for certain, but I would guess that you are passing in a mock which is raising a pytest exception. Why that exception is translated in to a `PytestUnknownMarkWarning` I couldn't say.

**Reply by sylvorg** [reply]
I am indeed using `mock`, but the problem no longer seems occurring... Interesting, very interesting. Either way, thanks for the help! I'll mark this as the answer for now!
